### DISCUSSION TITLE
select, where, relationship - postgresql+asyncpg.

### DISCUSSION THREAD

**Discussion by wnesbv** [sqlalchemy/sqlalchemy#8050] [Usage Questions]
Good day. There is a filter:

`        stmt = await session.execute(
            select(PersonParticipant.participant).where(
                PersonParticipant.request_group.has(
                    GroupChat.admin_group
                ) == request.user.user_id
            )
        )
        get_admin_group = stmt.scalars().first()
`

in sqlite3, everything works ok
postgresql+asyncpg:

sqlalchemy.exc.ProgrammingError: (sqlalchemy.dialects.postgresql.asyncpg.ProgrammingError) <class 'asyncpg.exceptions.DatatypeMismatchError'>:

[SQL: SELECT participant.participant
FROM participant
WHERE (EXISTS (SELECT 1
FROM groups_ch
WHERE groups_ch.id = participant.group_participant AND groups_ch.admin_group)) = %s]
[parameters: (2,)]

more detailed https://github.com/wnesbv/starlette-all/tree/main/participant

**Comment by CaselIT** [comment]
Hi,

The select you posted does not make sense, so pg seems right in returning an error.
I would guess that sqlite may execute the query but the result is not what you want.

What is the sql you want to execute?

**Reply by wnesbv** [reply]
Oh yeah.. Jumping from project to project - something stupidly copied and pasted..
